File: PYTHON/ENCAPSULAMENTO/funcionario .py

class Funcionario:
    def __init__(self,nome ,salario):
        self.nome = nome
        self.__salario = salario

# metodo getter:consulta o salario,pois esta privado
    def consultar_salario(self):
        return self.__salario

# metodo setter:modificaria para depositar o dinheiro com validaçao   
    def depositar(self, valor):
        if valor > 0:
            self.__saldo += valor
            print(f"Depósito de R$ {valor:.2f} realizado com sucesso.")
        else:
            print("O valor do depósito deve ser positivo!")
# encapsulamento:ocultar a implementação do cdigo ou seja serve 
# para proteger os dados internoos de uma classe

File: JAVA/ENCAPSULAMENTO/funcionario.java
package JAVA.ENCAPSULAMENTO;

public class Funcionario {
    String nome;
    private double salario;
    


    public Funcionario(String nome,double salario){
        this.nome = nome;
        this.salario = salario;

    }

    public double getSalario(){
        return this.salario;
    }

    public void setSalario(double novoSalario) {
        if (novoSalario > 0) {
            this.salario = novoSalario;
            System.out.println("Salário atualizado com sucesso.");
        } else {
            System.out.println("O valor do salário deve ser positivo!");
        }
    }
}
